java/src/org/openqa/selenium/remote/RemoteWebDriver.java

```diff
@@ -493,6 +497,76 @@ public Object executeAsyncScript(String script, Object... args) {
     return execute(DriverCommand.EXECUTE_ASYNC_SCRIPT(script, convertedArgs)).getValue();
   }
 
+  @Override
+  public ScriptKey pin(String script) {
+    UnpinnedScriptKey key = (UnpinnedScriptKey) JavascriptExecutor.super.pin(script);
+    String browserName = getCapabilities().getBrowserName().toLowerCase();
+    if ((browserName.equals("chrome") ||
+         browserName.equals("msedge") ||
+         browserName.equals("microsoftedge")) && this instanceof HasDevTools) {
+
+      ((HasDevTools) this).maybeGetDevTools().ifPresent(devTools -> {
+        devTools.createSessionIfThereIsNotOne();
+        devTools.send(new org.openqa.selenium.devtools.Command<>("Page.enable",
+                                                                 ImmutableMap.of()));
+        devTools.send(new org.openqa.selenium.devtools.Command<>("Runtime.evaluate",
+                                                                 ImmutableMap.of("expression",
+                                                                                 key.creationScript())));
+        Map<String, Object> result = devTools.send(new org.openqa.selenium.devtools.Command<>(
+          "Page.addScriptToEvaluateOnNewDocument",
+          ImmutableMap.of("source", key.creationScript()),
+          new TypeToken<Map<String, Object>>() {
+          }.getType()));
+        key.setScriptId((String) result.get("identifier"));
+      });
+    }
+    return key;
+  }
+
+  @Override
+  public void unpin(ScriptKey scriptKey) {
+    UnpinnedScriptKey key = (UnpinnedScriptKey) scriptKey;
+
+    JavascriptExecutor.super.unpin(key);
+
+    String browserName = getCapabilities().getBrowserName().toLowerCase();
+    if ((browserName.equals("chrome") ||
+         browserName.equals("msedge") ||
+         browserName.equals("microsoftedge")) && this instanceof HasDevTools) {
```

Comment:
Why do we need to check the browser name? If the driver `HasDevTools` it should be enough, right?

Reply:
Some of the CDP commands are not implemented in Firefox and that causes the functionality to break. 
Hence, the browser check.

Reply:
👍 
The plan is to move Firefox to BiDi as soon as possible?

Reply:
The plan is to add BiDi APIs that are implemented by browsers and support existing Selenium BiDi APIs as soon as we can.